Sources/Web3Modal/Core/Web3ModalClient.swift

```diff
@@ -157,10 +170,81 @@ public class Web3ModalClient {
+        case .wc:
+            guard
+                let session = getSessions().first,
+                let chain = getSelectedChain(),
+                let blockchain = Blockchain(namespace: chain.chainNamespace, reference: chain.chainReference)
+            else { return }
+            
+            if case let .personal_sign(address, message) = request {
+                try await signClient.request(
+                    params: .init(
+                        topic: session.topic,
+                        method: request.rawValues.method,
+                        params: AnyCodable(any: [message, address]),
+                        chainId: blockchain
+                    )
+                )
+            } else {
+                try await signClient.request(
+                    params: .init(
+                        topic: session.topic,
+                        method: request.rawValues.method,
+                        params: AnyCodable(any: request.rawValues.params),
+                        chainId: blockchain
+                    )
+                )
+            }
+        case .cb:
+                    
+            guard let jsonRpc = request.toCbAction() else { return }
+                    
+            // Execute on main as Coinbase SDK is not dispatching on main when calling UIApplication.openUrl()
+            DispatchQueue.main.async {
+                CoinbaseWalletSDK.shared.makeRequest(
+                    .init(
+                        actions: [
+                            Action(jsonRpc: jsonRpc)
+                        ]
+                    )
+                ) { result in
+                    let response: W3MResponse
+                    switch result {
+                    case let .success(payload):
+                        
+                        switch payload.content.first {
+                        case let .success(JSONString):
+                            response = .init(result: .response(AnyCodable(JSONString)))
+                        case let .failure(error):
+                            response = .init(result: .error(.init(code: error.code, message: error.message)))
+                        case .none:
+                            response = .init(result: .error(.init(code: -1, message: "Empty response")))
+                        }
+                    case let .failure(error):
+                        Web3Modal.config.onError(error)
+                        
+                        if let cbError = error as? ActionError {
+                            response = .init(result: .error(.init(code: cbError.code, message: cbError.message)))
+                        } else {
+                            response = .init(result: .error(.init(code: -1, message: error.localizedDescription)))
+                        }
+                    }
+                    
+                    self.coinbaseResponseSubject.send(response)
+                }
+            }
+        case .none:
```

Comment:
well it's just one way how to deal with `Optional`. So since I'm switching over the value anyway it felt more readable then some `guard` or `if let` 
```
enum Optional<Wrapped> {
    case none
    case some(Wrapped)
  }
  ```